library/iFixit/Matryoshka/KeyFix.php

```diff
@@ -12,31 +12,34 @@ class KeyFix extends KeyChange {
    const MD5_STRLEN = 32;
 
    private $maxLength;
-   private $invalidChars;
+
+   /** @var string */
+   private $invalidRegex;
 
    public function __construct(Backend $backend, $maxLength,
-    $invalidChars = " \n") {
+    $invalidRegex) {
       if ($maxLength < self::MD5_STRLEN) {
          throw new \InvalidArgumentException(
           'Max length must be larger than ' . self::MD5_STRLEN);
       }
 
-      if (!is_string($invalidChars) || $invalidChars === '') {
+      $isRegex = @preg_match($invalidRegex, '') !== false;
```

Comment:
This seems weird.

Reply:
Why not typehint $maxLength and $invalidRegex?

Reply:
regex is a string - `''` is not a valid regex + will cause preg_match to throw an exception. I wasn't sure what the best choice was here.

Reply:
I just mean is this something we need to be checking here? It'll hit this exception the first time it calls safekey right?

Reply:
And I meant typehinting the args to the constructor.

Reply:
We could just call `$this->safeString('')` and it'll throw or no-op.